cmd/podman/unshare.go

```diff
@@ -0,0 +1,63 @@
+// +build linux
+
+package main
+
+import (
+	"os"
+	"os/exec"
+
+	"github.com/containers/buildah/pkg/unshare"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	unshareDescription = "\n  Runs a command in a modified user namespace."
+	unshareCommand     = &cobra.Command{
+		Use:   "unshare",
+		Short: "Run a command in a modified user namespace",
+		Long:  unshareDescription,
+		RunE:  unshareCmd,
+		Example: `podman unshare id
+  podman unshare cat /proc/self/uid_map,
+  podman unshare podman-script.sh`,
+	}
+)
+
+func init() {
+	unshareCommand.SetUsageTemplate(UsageTemplate())
+	flags := unshareCommand.Flags()
+	flags.SetInterspersed(false)
+}
+
+// unshareCmd execs whatever using the ID mappings that we want to use for ourselves
+func unshareCmd(c *cobra.Command, args []string) error {
+	// Set the default isolation type to use the "rootless" method.
+	if _, present := os.LookupEnv("PODMAN_ISOLATION"); !present {
+		if err := os.Setenv("PODMAN_ISOLATION", "rootless"); err != nil {
+			logrus.Errorf("error setting PODMAN_ISOLATION=rootless in environment: %v", err)
+			os.Exit(1)
+		}
+	}
+
+	// force reexec using the configured ID mappings
+	unshare.MaybeReexecUsingUserNamespace(true)
+	// exec the specified command, if there is one
+	if len(args) < 1 {
+		// try to exec the shell, if one's set
+		shell, shellSet := os.LookupEnv("SHELL")
+		if !shellSet {
+			logrus.Errorf("no command specified")
+			os.Exit(1)
+		}
+		args = []string{shell}
+	}
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Env = unshare.RootlessEnv()
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	unshare.ExecRunnable(cmd)
+	os.Exit(1)
```

Comment:
why exit with a 1? And the return following?

Reply:
It exits with a 1 because `ExecRunnable()` exits with the exit status of the command, so if the control returns back to the function, it is definitely an error. The return nil is unnecessary I have now removed it.